### PR TITLE
[thread-cert] refactor case 6.2.2 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_6_2_02_NewPartition.py [hunk 1]
```diff
@@ -88,10 +94,37 @@ def test(self):
...
+        _ed_pkts.filter_mle_cmd(consts.MLE_ADVERTISEMENT).must_not_next()
+        _ed_pkts.filter_mle_cmd(consts.MLE_CHILD_ID_REQUEST).must_not_next()
+        _ed_pkts.filter_wpan_dst64(ROUTER_1).filter_mle_cmd(13).must_next(
+        ).must_verify(lambda p: {1, 0, 11} < set(p.mle.tlv.type))
+
+        # Step 8: The DUT MUST respond with ICMPv6 Echo Reply
+        ed_mleid = pv.vars['ED_MLEID']
+        router1_mleid = pv.vars['ROUTER_1_MLEID']
+        _ed_pkts.filter(lambda p: p.ipv6.dst == router1_mleid and p.ipv6.src ==
+                        ed_mleid).filter_ping_reply().must_next()
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_2_02_NewPartition.py:126]: nit: the code can be further simplified to:
```suggestion
        _ed_pkts.filter(lambda p: p.ipv6.dst == {ROUTER_1_MLEID} and p.ipv6.src ==
                        {ED_MLEID}, **pv.vars).filter_ping_reply().must_next()
```

**Comment by jwhui** [comment]
@FiveDimensions , please rebase and resolve comments above.

File: tests/scripts/thread-cert/Cert_6_2_02_NewPartition.py [hunk 2]
```diff
@@ -88,10 +94,34 @@ def test(self):
...
+
+        # Step 2-6 : remove Leader and make ROUTER_2 as new Leader
+        # Step 7: The DUT MUST send a MLE Child Update Request to Router_1
+        _ed_pkts.filter_mle_cmd(MLE_ADVERTISEMENT).must_not_next()
+        _ed_pkts.filter_mle_cmd(MLE_CHILD_ID_REQUEST).must_not_next()
+        _ed_pkts.filter_wpan_dst64(ROUTER_1).filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).must_next().must_verify(
+            lambda p: {MODE_TLV, SOURCE_ADDRESS_TLV, LEADER_DATA_TLV} < set(p.mle.tlv.type))
+
+        # Step 8: The DUT MUST respond with ICMPv6 Echo Reply
+        _ed_pkts.filter('ipv6.dst == {ROUTER_1_MLEID}' and 'ipv6.src == {ED_MLEID}',
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_6_2_02_NewPartition.py:122]:
```suggestion
        _ed_pkts.filter('ipv6.dst == {ROUTER_1_MLEID} and ipv6.src == {ED_MLEID}',
```

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 
Thanks!

**Pull request merged**